shard/key.go

```diff
@@ -30,6 +30,11 @@ func KeyFromCID(cid cid.Cid) Key {
 	return Key{str: cid.String()}
 }
 
+// KeyFromCIDMultihash returns a key based on the multihash of the given cid.
```

Comment:
I'm curious, why do you need this method?

Reply:
@dirkmc In RM, I wanna key all shards by their multihash i.e. if I have a CAR file for a given multihash, it dosen't matter what the codec is. This is just a helper method to create the shard key based on the multihash.